pulsar-broker/src/main/java/org/apache/pulsar/broker/delayed/bucket/BucketDelayedDeliveryTracker.java

```diff
@@ -611,7 +641,7 @@ public synchronized NavigableSet<Position> getScheduledMessages(int maxMessages)
             long ledgerId = sharedBucketPriorityQueue.peekN2();
             long entryId = sharedBucketPriorityQueue.peekN3();
 
-            ImmutableBucket bucket = snapshotSegmentLastIndexTable.get(ledgerId, entryId);
+            ImmutableBucket bucket = snapshotSegmentLastIndexMap.get(new SnapshotKey(ledgerId, entryId));
```

Comment:
it's worth assigning the SnapshotKey instance to a local variable so that a new instance doesn't need to be created later on line 670 and 680

Reply:
Agreed on the 644 and 670 line reference. But 680 seems to be a different ledger and entry.